sycl/source/detail/pi.cpp

```diff
@@ -435,47 +436,53 @@ std::vector<PluginPtr> &initialize() {
   return GlobalHandler::instance().getPlugins();
 }
 
+// Implementation of this function is OS specific. Please see windows_pi.cpp and
+// posix_pi.cpp.
+// TODO: refactor code when support matrix for DPCPP changes and <filesystem> is
+// available on all supported systems.
+std::vector<std::tuple<std::string, backend, void *>>
+loadPlugins(const std::vector<std::pair<std::string, backend>> &&PluginNames);
```

Comment:
Why not in `pi.hpp`?

Reply:
As @bader noticed, currently pi.hpp is in sycl/include/detail directory which means that it is exposed to the user. It seems like some of the functions from there need to be moved to the sycl/source/detail, because no need to expose them to the user, they used only internally.
This function is also used only internally, so I was trying to avoid adding this function to pi.hpp. And since it's only single usage, I didn't create header for it in sycl/source/detail.